cpp/src/strings/utilities.cuh

```diff
@@ -169,6 +170,130 @@ __host__ __device__ inline cudf::char_utf8 codepoint_to_utf8(uint32_t unchr)
   return utf8;
 }
 
+/**
```

Comment:
These utilities are specific to the converters. I would be more comfortable moving these into a new `cpp/src/strings/convert/utilties.cuh` file instead.